# HG changeset patch
# User jwe
# Date 1201679869 0
#      Wed Jan 30 07:57:49 2008 +0000
# Branch release-3-0-x
# Node ID 09f833d41f68455ff077e8ebbaa9282d8869b22e
# Parent  ccf8e0410ee0f87edaa6e81f690d142f1088f1e7
[3-0-0-branch @ 2008-01-30 07:57:48 by jwe]

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,14 @@
 2008-01-30  John W. Eaton  <jwe@octave.org>
 
+	* miscellaneous/tempdir.m: Append filesep to name for
+	compatibility.  Warn if not a directory or directory does not
+	exist.
+
 	* strings/deblank.m: Improve compatibility.
 
 2008-01-28  Michael Goffioul <michael.goffioul@gmail.com>
 
 	* plot/xlabel.m, plot/ylabel.m, plot/zlabel.m:
 	Return the label handle, not the current axis handle.
 
 2008-01-25  James Packer  <james.packer@wadh.ox.ac.uk>
diff --git a/scripts/miscellaneous/tempdir.m b/scripts/miscellaneous/tempdir.m
--- a/scripts/miscellaneous/tempdir.m
+++ b/scripts/miscellaneous/tempdir.m
@@ -19,13 +19,21 @@
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {@var{dir} =} fullfile ()
 ## Return the name of the system's directory for temporary files.
 ## @end deftypefn
 
 function dirname = tempdir ()
 
   dirname = getenv ("TMPDIR");
-  if (length (dirname) == 0)
+  if (isempty (dirname))
     dirname = P_tmpdir;
   endif
 
+  if (! strcmp (dirname(end), filesep))
+    strcat (dirname, filesep);
+  endif
+
+  if (! isdir (dirname))
+    warning ("tempdir: `%s' does not exist or is not a directory", dirname);
+  endif
+
 endfunction
